### PR TITLE
CLICKHOUSE-4018: copier: execute alter table drop partition with replication_alter_partitions_sync=1 (from settings)

### PR DESCRIPTION
**Pull request by proller**
…cation_alter_partitions_sync 1

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: dbms/programs/copier/ClusterCopier.cpp [hunk 1]
```diff
@@ -1277,7 +1277,7 @@ class ClusterCopier
         Settings settings_push = task_cluster->settings_push;
 
         /// It is important, DROP PARTITION must be done synchronously
```
Comment (velom) [inline dbms/programs/copier/ClusterCopier.cpp]: is it realy important?

File: dbms/programs/copier/ClusterCopier.cpp [hunk 2]
```diff
@@ -683,6 +683,11 @@ void DB::TaskCluster::loadTasks(const Poco::Util::AbstractConfiguration & config
     }
 }
 
+auto set_default_value = [] (auto && setting, auto && default_value)
+{
+        setting = setting.changed ? setting.value : default_value;
```
Comment (alexey-milovidov) [inline dbms/programs/copier/ClusterCopier.cpp]: .

File: dbms/programs/copier/ClusterCopier.cpp [hunk 3]
```diff
@@ -683,6 +683,11 @@ void DB::TaskCluster::loadTasks(const Poco::Util::AbstractConfiguration & config
     }
 }
 
+auto set_default_value = [] (auto && setting, auto && default_value)
```
Comment (alexey-milovidov) [inline dbms/programs/copier/ClusterCopier.cpp]: Global lambda.

File: dbms/src/Interpreters/Settings.h [hunk 4]
```diff
@@ -335,6 +335,12 @@ struct Settings
 
     /// Dumps profile events to two columns of type Array(String)
     void dumpToArrayColumns(IColumn * column_names, IColumn * column_values, bool changed_only = true);
+
+    template <typename SETTING, typename TYPE>
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/Settings.h]: Missing comment.

**Comment by alexey-milovidov** [comment]
Non obvious.

**Comment by alexey-milovidov** [comment]
We have not concluded whether this possibility safe and whether it makes any sense.

**Pull request closed without merging**